modules/monitoring/templates/smokey-loop.conf

```diff
@@ -1,9 +1,19 @@
 description "Smokey loop outputs JSON which is consumed by monitoring"
 
+setuid smokey
+setgid smokey
+
 # Run in a continuous loop. `respawn` catches non-zero exit codes. Whereas
 # `and stopped` catches normal exits. The default `respawn limit` will
 # prevent this from spinning if the script is broken.
 start on runlevel [2345] and stopped $UPSTART_JOB
 respawn
+respawn limit unlimited
+
+kill timeout 240
+
+exec /opt/smokey/tests_json_output.sh /tmp/smokey.json.tmp integration
 
-exec sudo -u smokey /opt/smokey/tests_json_output.sh /tmp/smokey.json <%= @environment %>
+post-stop script
+    mv /tmp/smokey.json.tmp /tmp/smokey.json
```

Comment:
Why don't we output to `/tmp/smokey.json` straight away in line 15?

Reply:
I think that cucumber might clear the file when it starts. The purpose of writing a JSON file out is so that Icinga can check the contents, so really you want it to contain useful information most of the time. Does that make sense?

Reply:
Yep - so otherwise, Icinga would not be able to check its contents while it's running. Having it as a tmp file means the results are always available 👍